dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -105,30 +105,41 @@ def refute_estimate(self):
         self.logger.info("Refutation over {} simulated datasets".format(self._num_simulations) )
         self.logger.info("The transformation passed: {}".format(self._transformations) )
 
-        data_chunks = self.preprocess_data_by_treatment()
+        groups = self.preprocess_data_by_treatment()
         estimates = []
-        for chunk in data_chunks:
-
-            X_input = chunk[self._chosen_variables]
-            new_outcome = chunk['y']
-            X = self._data[self._chosen_variables]
+        for key_train, _ in groups:
+            X_train = groups.get_group(key_train)[self._chosen_variables].values
+            new_outcome_train = groups.get_group(key_train)['y'].values
+            validation_df = [] 
+            for key_validation, _ in groups:
+                if key_validation != key_train:
+                    validation_df.append(groups.get_group(key_validation))
+
+            validation_df = pd.concat(validation_df)
+            X_validation = validation_df[self._chosen_variables].values
+            new_outcome_validation = validation_df['y'].values
 
             for action, func_args in self._transformations:
 
                 if callable(action):
-                    estimator = action(X_input, new_outcome, **func_args)
-                    new_outcome = estimator(X)
+                    estimator = action(X_train, new_outcome_train, **func_args)
+                    new_outcome_train = estimator(X_train)

```

Comment:
do we need to estimate and store the "new_outcome_train"? is it used anywhere?

Reply:
The point of having a new_outcome_train, is to store the value temporarily as the input for another part of the transformation list.
For example, if we first perform a permutation followed by an estimation. We need to perform the estimation on the permuted data.